From 2aad72978df12b0d7e39b5f857d1515e51e73026 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:00 -0400
Subject: [acpi] hotplug/pci: Simplify register_slot()

Message-id: <1411129776-21656-26-git-send-email-prarit@redhat.com>
Patchwork-id: 96352
O-Subject: [RHEL7.1 PATCH BZ 1141846 25/61] ACPI / hotplug / PCI: Simplify register_slot()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 146fc68a4bdd78e49d56f1530f6b8072034cf3ef
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Tue Feb 4 00:38:15 2014 +0100

    ACPI / hotplug / PCI: Simplify register_slot()

    The err label in register_slot() is only jumped to from one place,
    so move the code under the label to that place and drop the label.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Tested-by: Mika Westerberg <mika.westerberg@linux.intel.com>

Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 633e394..e14e0b1 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -335,8 +335,10 @@ static acpi_status register_slot(acpi_handle handle, u32 lvl, void *data,
 
  slot = kzalloc(sizeof(struct acpiphp_slot), GFP_KERNEL);
  if (!slot) {
-  status = AE_NO_MEMORY;
-  goto err;
+  mutex_lock(&acpiphp_context_lock);
+  acpiphp_put_context(context);
+  mutex_unlock(&acpiphp_context_lock);
+  return AE_NO_MEMORY;
  }
 
  slot->bus = bridge->pci_bus;
@@ -404,12 +406,6 @@ static acpi_status register_slot(acpi_handle handle, u32 lvl, void *data,
  }
 
  return AE_OK;
-
- err:
- mutex_lock(&acpiphp_context_lock);
- acpiphp_put_context(context);
- mutex_unlock(&acpiphp_context_lock);
- return status;
 }
 
 static struct acpiphp_bridge *acpiphp_handle_to_bridge(acpi_handle handle)
-- 
1.7.1